components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/nats/CoreListener.java

```diff
@@ -0,0 +1,304 @@
+        String tlsProtocol = validateParameter(natsProperties.getProperty(NatsConstants.TLS_PROTOCOL));
+        String tlsKeyStoreType = validateParameter(natsProperties.getProperty(NatsConstants.TLS_KEYSTORE_TYPE));
+        String tlsKeyStoreLocation = validateParameter(natsProperties.getProperty(NatsConstants.TLS_KEYSTORE_LOCATION));
+        String tlsKeyStorePassword = validateParameter(natsProperties.getProperty(NatsConstants.TLS_KEYSTORE_PASSWORD));
+        String tlsTrustStoreType = validateParameter(natsProperties.getProperty(NatsConstants.TLS_TRUSTSTORE_TYPE));
+        String tlsTrustStoreLocation = validateParameter(
+                natsProperties.getProperty(NatsConstants.TLS_TRUSTSTORE_LOCATION));
+        String tlsTrustStorePassword = validateParameter(
+                natsProperties.getProperty(NatsConstants.TLS_TRUSTSTORE_PASSWORD));
+        String tlsKeyManagerAlgorithm = validateParameter(
+                natsProperties.getProperty(NatsConstants.TLS_KEY_MANAGER_ALGORITHM));
+        String tlsTrustManagerAlgorithm = validateParameter(
+                natsProperties.getProperty(NatsConstants.TLS_TRUST_MANAGER_ALGORITHM));
+
+        Options.Builder builder = new Options.Builder(natsProperties);
+
+        if (StringUtils.isNotEmpty(bufferSize)) {
+            builder.bufferSize(Integer.parseInt(bufferSize));
+        }
+
+        if (Boolean.parseBoolean(turnOnAdvancedStats)) {
+            builder.turnOnAdvancedStats();
+        }
+
+        if (Boolean.parseBoolean(traceConnection)) {
+            builder.traceConnection();
+        }
+
+        if (StringUtils.isNotEmpty(
+                tlsProtocol + tlsTrustStoreType + tlsTrustStoreLocation + tlsTrustStorePassword + tlsKeyStoreType
+                        + tlsKeyStoreLocation + tlsKeyStorePassword + tlsKeyManagerAlgorithm
+                        + tlsTrustManagerAlgorithm)) {
+            SSLContext sslContext = createSSLContext(
+                    new TLSConnection(tlsProtocol, tlsTrustStoreType, tlsTrustStoreLocation, tlsTrustStorePassword,
+                            tlsKeyStoreType, tlsKeyStoreLocation, tlsKeyStorePassword, tlsKeyManagerAlgorithm,
+                            tlsTrustManagerAlgorithm));
+            if (sslContext != null) {
+                builder.sslContext(sslContext);
+            }
+        }
+        return Nats.connect(builder.build());
+    }
+
+    /**
+     * Consume the message received and inject into the sequence.
+     *
+     * @param sequenceName the sequence to inject the message to.
+     */
+    @Override public void consumeMessage(String sequenceName) {
+        Dispatcher dispatcher;
+        dispatcher = connection.createDispatcher(natsMessage -> {
+            if (natsMessage != null) {
+                String message = new String(natsMessage.getData(), StandardCharsets.UTF_8);
+                log.info("Message Received to NATS Inbound EP: " + message);
```

Comment:
Redundant info log. We don't need to print the message as info anyway, since it could contain sensitive data, or be a large message